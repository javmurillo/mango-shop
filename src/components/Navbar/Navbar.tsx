import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import logo from '../../logo.svg';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

const StyledLogo = styled.img`
  height: 25px;
`;

export const MangoNavbar = () => {
  return (
    <Navbar bg="light" expand="lg">
      <Navbar.Brand>
        <a href="https://shop.mango.com/" target="_blank" rel="noreferrer">
          <StyledLogo src={logo} alt="logo" />
        </a>
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <Nav className="mr-auto">
          <ul className="navbar-nav">
            <li className="nav-item">
              <NavLink
                activeClassName="active"
                className="nav-link"
                to="/exercise1"
                exact
              >
                Normal range
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink
                activeClassName="active"
                className="nav-link"
                to="/exercise2"
                exact
              >
                Fixed values range
              </NavLink>
            </li>
          </ul>
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
};
