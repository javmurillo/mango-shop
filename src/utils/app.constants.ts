export const API_URL = 'https://demo4557431.mockable.io';
