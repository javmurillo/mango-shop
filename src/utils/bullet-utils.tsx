import { RangeBulletProps } from '../components/RangeBullet/range-bullet-props.interface';
import { RangeBulletState } from '../components/RangeBullet/range-bullet-state.interface';
import { BulletStyles } from '../components/RangeBullet/styles';

export function getBulletStyle(
  styles: BulletStyles,
  state: RangeBulletState,
  props: RangeBulletProps
) {
  const { hovered, focused, active } = state;
  const { offset } = props;
  let calcHoverStyle;
  let calcFocusStyle;
  let calcActiveStyle;

  if (hovered) {
    calcHoverStyle = { ...styles.hoveredHandle };
  }
  if (focused) {
    calcFocusStyle = { ...styles.focusedHandle };
  }
  if (active) {
    calcActiveStyle = { ...styles.activeHandle };
  }

  return {
    ...styles.handle,
    left: offset,
    ...calcHoverStyle,
    ...calcFocusStyle,
    ...calcActiveStyle,
  };
}
