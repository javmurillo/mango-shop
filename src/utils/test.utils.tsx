// setupTests.tsx

import React from 'react';
import { Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { createMemoryHistory, MemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { store } from '../store/app.store';

interface RenderWithRouterProps {
  route?: string;
  history?: MemoryHistory;
}

export const renderWithProviderAndRouter = (
  ui: React.ReactNode,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  }: RenderWithRouterProps = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>{ui}</Router>
      </Provider>
    ),
    history,
  };
};
