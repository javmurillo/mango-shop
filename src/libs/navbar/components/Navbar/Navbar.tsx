import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';
import mangoLogo from '../../../../logo.svg';

const StyledLogo = styled.img`
  height: 25px;
`;

/**
 * Navbar component
 */
export const MangoNavbar = (): JSX.Element => {
  return (
    <Navbar bg="light" expand="lg" aria-label="navbar">
      <Navbar.Brand>
        <a href="https://shop.mango.com/" target="_blank" rel="noreferrer">
          <StyledLogo src={mangoLogo} alt="logo" aria-label="mango logo" />
        </a>
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <Nav className="mr-auto">
          <ul className="navbar-nav">
            <li className="nav-item">
              <NavLink
                aria-label="exercise1"
                activeClassName="active"
                className="nav-link"
                to="/exercise1"
                exact
              >
                Normal range
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink
                aria-label="exercise2"
                activeClassName="active"
                className="nav-link"
                to="/exercise2"
                exact
              >
                Fixed values range
              </NavLink>
            </li>
          </ul>
        </Nav>
      </Navbar.Collapse>
      <Navbar.Brand>
        <a
          href="https://github.com/javmurillo/mango-shop"
          target="_blank"
          rel="noreferrer"
        >
          <StyledLogo
            src={process.env.PUBLIC_URL + '/github-logo.png'}
            alt="logo"
            aria-label="github logo"
          />
        </a>
      </Navbar.Brand>
    </Navbar>
  );
};
