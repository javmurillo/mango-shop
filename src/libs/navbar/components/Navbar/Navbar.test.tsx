import { cleanup, render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import { renderWithProviderAndRouter } from '../../../../utils/test.utils';
import { MangoNavbar } from './Navbar';

afterEach(cleanup);

const setup = () => {
  const utils = renderWithProviderAndRouter(
    <MangoNavbar aria-label="navbar" />
  );
  const navbar = utils.getByLabelText('navbar');
  return {
    navbar,
    ...utils,
  };
};

describe('<Navbar/> tests', () => {
  test('renders correctly', () => {
    const { navbar } = setup();
    expect(navbar).toBeInTheDocument();
  });

  test('shoud match snapshot', () => {
    const testRendererJson = TestRenderer.create(
      <Router history={createMemoryHistory({ initialEntries: ['/'] })}>
        <MangoNavbar aria-label="navbar" />
      </Router>
    ).toJSON();
    expect(testRendererJson).toMatchSnapshot();
  });

  test('mango logo is present', () => {
    setup();
    const mangoLogo = screen.getByLabelText('mango logo');
    expect(mangoLogo).toBeInTheDocument();
  });

  test('github logo is present', () => {
    setup();
    const mangoLogo = screen.getByLabelText('github logo');
    expect(mangoLogo).toBeInTheDocument();
  });
});
