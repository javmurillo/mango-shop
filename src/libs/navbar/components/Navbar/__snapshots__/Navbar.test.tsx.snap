// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar/> tests shoud match snapshot 1`] = `
<nav
  aria-label="navbar"
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <span
    className="navbar-brand"
  >
    <a
      href="https://shop.mango.com/"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="logo"
        aria-label="mango logo"
        className="sc-bdfBwQ WWxNV"
        src="logo.svg"
      />
    </a>
  </span>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            aria-label="exercise1"
            className="nav-link"
            href="/exercise1"
            onClick={[Function]}
          >
            Normal range
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            aria-label="exercise2"
            className="nav-link"
            href="/exercise2"
            onClick={[Function]}
          >
            Fixed values range
          </a>
        </li>
      </ul>
    </div>
  </div>
  <span
    className="navbar-brand"
  >
    <a
      href="https://github.com/javmurillo/mango-shop"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="logo"
        aria-label="github logo"
        className="sc-bdfBwQ WWxNV"
        src="/github-logo.png"
      />
    </a>
  </span>
</nav>
`;
